src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -143,5 +153,32 @@ trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
     )
     response
   }
+
+  def autocompleteSuggestions(client: TransportClient, indexname: String, criteria: LibrarySearchParams): Future[LibrarySearchResponse] = {
+
+    val searchQuery = buildAutocompleteQuery(client, indexname, criteria)
+
+    logger.debug(s"autocomplete search query: $searchQuery.toJson")
+    val searchFuture = Future[SearchResponse](executeESRequest[SearchRequest, SearchResponse, SearchRequestBuilder](searchQuery))
+
+    searchFuture map {searchResult =>
+
+      // autocomplete query can return duplicate suggestions. De-dupe them here.
+      val suggestions:List[JsString] = (searchResult.getHits.getHits.toList flatMap { hit =>
+        if (hit.getHighlightFields.containsKey(fieldSuggest)) {
+          hit.getHighlightFields.get(fieldSuggest).fragments map {t => JsString(t.toString)}
+        } else {
+          None
+        }
+      }).distinct
```

Comment:
As mentioned in another comment, this is case sensitive and should likely be case-insensitive to clarify the number of results.